calico_rkt/calico_policy.py

```diff
@@ -0,0 +1,199 @@
+        """
+
+        self.cni_path = env[CNI_PATH_ENV]
+        """
+        Path in which to search for CNI plugins.
+        """
+
+        self.network_name = network_config["name"]
+        """
+        Name of the network from the provided network config file.
+        """
+
+    def parse_cni_args(self, cni_args):
+        """Parses the given CNI_ARGS string into key value pairs
+        and returns a dictionary containing the arguments.
+
+        e.g "FOO=BAR;ABC=123" -> {"FOO": "BAR", "ABC": "123"}
+
+        :param cni_args
+        :return: args_to_return - dictionary of parsed cni args
+        """
+        # Dictionary to return.
+        args_to_return = {}
+
+        # For each arg, split at the equal sign and add to the dictionary.
+        _log.debug("Parsing CNI_ARGS: %s", cni_args)
+        for arg in cni_args.split(";"):
+            _log.debug("\tParsing CNI_ARG: %s", arg)
+            k, v = arg.split("=")
+            args_to_return[k] = v
+        return args_to_return
+
+    def execute(self, endpoint):
+        """Use CNI_COMMAND value to execute the appropriate command.
+
+        If CNI_COMMAND is ADD, call set_profile.
+        If CNI_COMAND is DEL, call remove_profile
+
+        :return: None.
+        """
+        if self.command == CNI_CMD_ADD:
+            self.set_profile(endpoint)
+        elif self.command == CNI_CMD_DELETE:
+            self.remove_profile()
+
+    def set_profile(self, endpoint):
+        """Sets a profile for the networked container on the given endpoint.
+
+        Create a profile if needed.
+
+        :param endpoint:
+        :return:
+        """
+        raise NotImplementedError("Must implement method in subclass")
+
+    def remove_profile(self):
+        """
+        Removes a profile for the networked container.
+
+        :return:
+        """
+        raise NotImplementedError("Must implement method in subclass")
+
+
+class DefaultPolicyPlugin(BasePolicyPlugin):
+    def set_profile(self, endpoint):
+        """Sets profile using the network name on given endpoint.
+
+        Default rules are allow only from containers in same profile.
+
+        If profile for the network name does not exist, create a profile.
+        Network name is set in the network config file.
+
+        :param endpoint
+        :return: None.
+        """
+        profile_name = self.network_name
+        if not self._client.profile_exists(profile_name):
+            # If the profile doesn't exist, create it.
+            _log.info("Creating new profile '%s'", profile_name)
+            self._client.create_profile(profile_name)
+
+        # Set the default profile on this pod's Calico endpoint.
+        _log.info("Setting profile %s on endpoint.", profile_name)
+        self._client.set_profiles_on_endpoint(profile_names=[profile_name],
+                                              endpoint_id=endpoint.endpoint_id)
+
+    def remove_profile(self):
+        """Remove the profile from datastore.
```

Comment:
Docstring on this guy is wrong - it should explain why we aren't doing anything here.